src/sentences/wnc.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{parse::{NmeaSentence, TEXT_PARAMETER_MAX_LEN}, Error, SentenceType};
+use arrayvec::ArrayString;
+use nom::{
+    character::complete::char,
+    bytes::complete::is_not,
+    combinator::opt, 
+    number::complete::float
+};
+use super::utils::array_string;
+
+/// WNC - Distance, Waypoint to Waypoint
+#[derive(Debug, PartialEq)]
+pub struct WncData {
+    /// Distance, Nautical Miles
+    pub distance_nautical_miles: Option<f32>,
+    /// Distance, Kilometers
+    pub distance_kilometers: Option<f32>,
+    /// Waypoint ID, Destination
+    pub waypoint_id_destination: Option<ArrayString<TEXT_PARAMETER_MAX_LEN>>,
+    /// Waypoint ID, Origin
+    pub waypoint_id_origin: Option<ArrayString<TEXT_PARAMETER_MAX_LEN>>,
+}
+
+/// Parses the WNC sentence into a data structure
+///
+/// Example of WNC sentences:
+/// - $GPWNC,12.3,N,45.6,K,DEST,START*hh
```

Comment:
Would you mind adding docs about the fields and also a link to https://gpsd.gitlab.io/gpsd/NMEA.html#_wnc_distance_waypoint_to_waypoint ?

E.g.:

https://github.com/AeroRust/nmea/blob/db5c7d80e2b5766463b04b7855e809a520325a16/src/sentences/rmz.rs#L20-L35